sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -617,6 +618,67 @@ export class NotificationActionsService {
     );
   }
 
+  /**
+   * Creates e-cert blocked notification for student and ministry.
+   * @param notification notification details.
+   * @param auditUserId user who blocks the ecert.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  async saveEcertBlockedNotification(
+    notification: EcertBlockedNotification,
+    auditUserId: number,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const studentNotificationTemplateId =
+      await this.notificationMessageService.getTemplateId(
+        NotificationMessageType.StudentNotificationEcertBlocked,
+      );
+    const studentNotificationToSend = {
+      userId: auditUserId,
+      messageType: NotificationMessageType.StudentNotificationEcertBlocked,
+      messagePayload: {
+        email_address: notification.email,
+        template_id: studentNotificationTemplateId,
+        personalisation: {
+          givenNames: notification.givenNames ?? "",
+          lastName: notification.lastName,
+        },
+      },
+    };
+    const ministryNotificationTemplateId =
+      await this.notificationMessageService.getTemplateId(
+        NotificationMessageType.MinistryNotificationEcertBlocked,
+      );
+    const ministryNotificationToSend = {
+      userId: auditUserId,
+      messageType: NotificationMessageType.MinistryNotificationEcertBlocked,
+      messagePayload: {
+        email_address: this.gcNotifyService.ministryToAddress(),
+        template_id: ministryNotificationTemplateId,
+        personalisation: {
+          givenNames: notification.givenNames ?? "",
+          lastName: notification.lastName,
+          dob: getDateOnlyFormat(notification.dob),
+          studentEmail: notification.email,
+          applicationNumber: notification.applicationNumber,
+          dateTime: this.getDateTimeOnPSTTimeZone(),
```

Comment:
Why are we required to show current date time in email ?? I couldn't sense any use of this. What is business need here??

Reply:
There was a business request in the past and many, if not all, templates follow the same.
I can see at least other 13 references to the same. The idea is probably to capture the moment when the event happened.